pkg/framework/context/context_test.go

```diff
@@ -132,19 +134,25 @@ func TestNewProvider(t *testing.T) {
 		require.Contains(t, err.Error(), "error handling the message")
 	})
 
-	t.Run("test new with generic inbound service", func(t *testing.T) {
+	t.Run("test new with message service", func(t *testing.T) {
 		const sampleMsgType = "generic-msg-type-2.0"
 
 		handled := make(chan bool, 1)
-		prov, err := New(WithMessageServices(&generic.MockGenericSvc{
+
+		mockMsgHandler := msghandler.NewMockMessageHandler()
```

Comment:
Isn't this a mock MessageServiceProvider?

Reply:
It is both actually, but renamed it MessageServiceProvider. This mock also implements Register/Unregister actually.